src/us/kbase/assemblyhomology/filters/KBaseAuthenticatedFilterFactory.java

```diff
@@ -0,0 +1,191 @@
+import us.kbase.workspace.WorkspaceClient;
+
+/** A factory for an authenticated KBase filter. The filter, once built, inspects the sequence
+ * IDs in the provided {@link MinHashDistance} instances
+ * (see {@link KBaseAuthenticatedFilter#accept(MinHashDistance)}
+ * and determines if the user identified by the supplied token is authorized to view the sequence.
+ * If not, the filter does not pass the distance on to the collector.
+ * 
+ * At the time of filter creation, the KBase workspace is contacted to get the workspace IDs to
+ * which the user has access, including public workspaces. The filter expects sequence IDs in the
+ * format X_Y_Z, where X is the integer workspace ID, Y the integer object ID, and Z the integer
+ * version.
+ * @author gaprice@lbl.gov
+ *
+ */
+public class KBaseAuthenticatedFilterFactory implements MinHashDistanceFilterFactory {
+
+	//note this is tested in an integration test as it requires the KBase workspace to function.
+
+	private static final List<String> ENVS = Collections.unmodifiableList(Arrays.asList(
+			"prod", "appdev", "next", "ci"));
+	
+	private static final String CONFIG_ENV = "env";
+	private static final String CONFIG_URL = "url";
+	
+	private final FilterID id;
+	private final URL url;
+	private final boolean insecure;
+	
+	/** Create the factory. The factory accepts two configuration parameters - 'url', which must
+	 * be the url of a KBase workspace service, and 'env', which identifies the KBase environment
+	 * (one of prod, appdev, next, or ci) the filter is associated with. This allows for
+	 * a single Assembly Homology service to be used for multiple KBase environments, if
+	 * desired. If omitted, the default of 'prod' is used.
+	 * @param config the filter configuration.
+	 * @throws MinHashFilterFactoryInitializationException if an initialization error occurs.
+	 */
+	public KBaseAuthenticatedFilterFactory(final Map<String, String> config)
+			throws MinHashFilterFactoryInitializationException {
+		checkNotNull(config, "config");
+		final String env;
+		if (config.containsKey(CONFIG_ENV)) {
+			env = config.get(CONFIG_ENV);
+			if (!ENVS.contains(env)) {
+				throw new MinHashFilterFactoryInitializationException(
+						"Illegal KBase filter environment value: " + env);
+			}
+		} else {
+			env = ENVS.get(0);
+		}
+		try {
+			id = new FilterID("kbase" + env);
+		} catch (MissingParameterException | IllegalParameterException e) {
+			throw new RuntimeException("This should be impossible", e);
+		}
+		url = getURL(config);
+		if (!url.getProtocol().equals("https")) {
+			insecure = true;
+			// logging is checked manually. Check that logging occurs if you make changes here
+			LoggerFactory.getLogger(getClass()).info(String.format(
+					"Workspace url %s is insecure. It is strongly recommended to use https.",
+					url));
+		} else {
+			// can't really test this easily
+			insecure = false;
+		}
+	}
+
+	private URL getURL(final Map<String, String> config)
```

Comment:
I think it would be useful to have a comment for this method or to name it more specifically,  since getURL makes me think of a normal getter, when in fact it does 
1) Gets url for KBASE Filter
2) checks to see if it is a valid URL, and 
3) requires that the URL be able to be used for instantiating a workspace client

Reply:
That function call now looks like this in my local repo:
```
url = validateWorkspaceURL(config.get(CONFIG_WS_URL));
```

It'll show up in a future PR.

Reply:
Function sig:
```
	private URL validateWorkspaceURL(final String putativeURL)
			throws MinHashFilterFactoryInitializationException {
```